contracts/BazrToken.sol

```diff
@@ -30,4 +30,9 @@ contract BazrToken is Initializable, ERC20UpgradeSafe {
         require(msg.sender == factory, "Only callable by Factory");
         vault = _vault;
     }
+
+    function _transfer(address sender, address recipient, uint256 amount) internal override {
+        require(sender == vault || recipient == vault, "Only allow to transfer to or from Vault");
+        super._transfer(sender, recipient, amount);
+    }
```

Comment:
This looks good but would still allow users to `transfer` or `transferFrom` to the Vault without going through `withdraw`. Not sure what would compel them to do this, but it would mess up the supply if they do, and the bTokens will just be stuck in the Vault.

I propose we override the external `transfer` and `transferFrom` functions to only let them be called by the Vault with `require(msg.sender == vault)`

Reply:
> This looks good but would still allow users to transfer or transferFrom to the Vault without going through withdraw.

very good point

I will move the require statement to external transfer and transferFrom, this should give us what we need